commit be365e5418d929760bdfed57404eb313fe9dba4b
Author: Mark Emlyn David Thomas <markt@apache.org>
Date:   Thu May 10 21:07:22 2012 +0000

    Fix https://issues.apache.org/bugzilla/show_bug.cgi?id=53063
    When using an Executor with BIO, use the executor's maxThreads as the default for maxConnections
    
    git-svn-id: https://svn.apache.org/repos/asf/tomcat/trunk@1336884 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/java/org/apache/tomcat/util/net/AbstractEndpoint.java b/java/org/apache/tomcat/util/net/AbstractEndpoint.java
index 30e34ed..a39aee0 100644
--- a/java/org/apache/tomcat/util/net/AbstractEndpoint.java
+++ b/java/org/apache/tomcat/util/net/AbstractEndpoint.java
@@ -305,7 +305,10 @@ public abstract class AbstractEndpoint {
         }
     }
     public int getMaxThreads() {
-        if (running && executor!=null) {
+        return getMaxThreadsExecutor(running);
+    }
+    protected int getMaxThreadsExecutor(boolean useExecutor) {
+        if (useExecutor && executor != null) {
             if (executor instanceof java.util.concurrent.ThreadPoolExecutor) {
                 return ((java.util.concurrent.ThreadPoolExecutor)executor).getMaximumPoolSize();
             } else if (executor instanceof ResizableExecutor) {
diff --git a/java/org/apache/tomcat/util/net/JIoEndpoint.java b/java/org/apache/tomcat/util/net/JIoEndpoint.java
index cb61fd3..6f1f1a0 100644
--- a/java/org/apache/tomcat/util/net/JIoEndpoint.java
+++ b/java/org/apache/tomcat/util/net/JIoEndpoint.java
@@ -376,7 +376,7 @@ public class JIoEndpoint extends AbstractEndpoint {
         // Initialize maxConnections
         if (getMaxConnections() == 0) {
             // User hasn't set a value - use the default
-            setMaxConnections(getMaxThreads());
+            setMaxConnections(getMaxThreadsExecutor(true));
         }
 
         if (serverSocketFactory == null) {
diff --git a/webapps/docs/config/ajp.xml b/webapps/docs/config/ajp.xml
index a06e8f9..d8205cd 100644
--- a/webapps/docs/config/ajp.xml
+++ b/webapps/docs/config/ajp.xml
@@ -345,7 +345,9 @@
       falls below this value. The operating system may still accept
       connections based on the <code>acceptCount</code> setting. Default value
       varies by connector type. For BIO the default is the value of
-      <strong>maxThreads</strong>. For NIO the default is <code>10000</code>.
+      <strong>maxThreads</strong> unless an <a href="executir.html">Executor</a>
+      is used in which case the default will be the value of maxThreads from the
+      executor. For NIO the default is <code>10000</code>.
       For APR/native, the default is <code>8192</code>.</p>
       <p>Note that for APR/native on Windows, the configured value will be
       reduced to the highest multiple of 1024 that is less than or equal to
diff --git a/webapps/docs/config/http.xml b/webapps/docs/config/http.xml
index f230a29..8dfd4a3 100644
--- a/webapps/docs/config/http.xml
+++ b/webapps/docs/config/http.xml
@@ -380,7 +380,9 @@
       falls below this value. The operating system may still accept
       connections based on the <code>acceptCount</code> setting. Default value
       varies by connector type. For BIO the default is the value of
-      <strong>maxThreads</strong>. For NIO the default is <code>10000</code>.
+      <strong>maxThreads</strong> unless an <a href="executir.html">Executor</a>
+      is used in which case the default will be the value of maxThreads from the
+      executor. For NIO the default is <code>10000</code>.
       For APR/native, the default is <code>8192</code>.</p>
       <p>Note that for APR/native on Windows, the configured value will be
       reduced to the highest multiple of 1024 that is less than or equal to
